Helper/CheckoutAnalytics.php

```diff
@@ -0,0 +1,297 @@
+            if (isset($message[$key])) {
+                $request[$key] = $message[$key];
+                $isMessageParamAdded = true;
+            }
+        }
+
+        if (!$isMessageParamAdded) {
+            throw new InvalidArgumentException(__('Message does not contain required fields!'));
+        }
+
+        return $request;
+    }
+
+    /**
+     * Generates the endpoint URL for initializing the checkout attempt
+     *
+     * @return string
+     * @throws AdyenException
+     * @throws NoSuchEntityException
+     */
+    private function getInitiateAnalyticsUrl(): string
+    {
+        $storeId = $this->storeManager->getStore()->getId();
+        $isDemoMode = $this->configHelper->isDemoMode($storeId);
+        $mode = $isDemoMode ? 'test' : 'live';
+        $clientKey = $this->configHelper->getClientKey($mode, $storeId);
+
+        if (is_null($clientKey)) {
+            throw new AdyenException("clientKey is not configured!");
+        }
+
+        return sprintf("%s?clientKey=%s", $this->getEndpointUrl($isDemoMode), $clientKey);
+    }
+
+    /**
+     * Builds the request array for initiate checkout attempt
+     * For extra fields, see constant EXTRA_PARAMS_INIT_ENDPOINT.
+     *
+     * @param array $extraParams
+     * @return array
+     * @throws NoSuchEntityException
+     */
+    private function buildInitiateCheckoutRequest(array $extraParams = []): array
+    {
+        $storeId = $this->storeManager->getStore()->getId();
+        $platformData = $this->adyenHelper->getMagentoDetails();
+        $storeLocale = $this->adyenHelper->getStoreLocale($storeId);
+        $mappedLocale = $this->locale->mapLocaleCode($storeLocale);
+        $url = $this->urlHelper->getCurrentUrl();
+
+        $request = [
+            'locale' => $mappedLocale,
+            'flavor' => self::FLAVOR_COMPONENT,
+            'referrer' => $url,
+            'applicationInfo' => [
+                'merchantApplication' => [
+                    'name' => $this->adyenHelper->getModuleName(),
+                    'version' => $this->adyenHelper->getModuleVersion()
+                ],
+                'externalPlatform' => [
+                    'name' => $platformData['name'],
+                    'version' => $platformData['version'],
+                    'integrator' => self::INTEGRATOR_ADYEN
+                ]
+            ]
+        ];
+
+        foreach (self::EXTRA_PARAMS_INIT_ENDPOINT as $key) {
+            if (array_key_exists($key, $extraParams)) {
+                $request[$key] = $extraParams[$key];
+            }
+        }
+
+        return $request;
+    }
+
+    /**
+     * @param $response
+     * @return bool
+     * @throws InvalidArgumentException
+     */
+    private function validateInitiateCheckoutAttemptResponse($response): bool
+    {
+        if(!array_key_exists('checkoutAttemptId', $response)) {
+            throw new InvalidArgumentException(__('checkoutAttemptId is missing in the response!'));
```

Comment:
How about logging an error here and return `false`. So that this does not interrupt the flow for the shopper